pkg/distributors/contracts/MultiDistributor.sol

```diff
@@ -701,30 +710,30 @@ contract MultiDistributor is IMultiDistributor, ReentrancyGuard, MultiDistributo
         returns (uint256 updatedGlobalTokensPerStake)
     {
         updatedGlobalTokensPerStake = _globalTokensPerStake(distribution);
-        distribution.globalTokensPerStake = updatedGlobalTokensPerStake;
-        distribution.lastUpdateTime = _lastTimePaymentApplicable(distribution);
+        distribution.globalTokensPerStake = uint192(updatedGlobalTokensPerStake);
+        distribution.lastUpdateTime = uint64(_lastTimePaymentApplicable(distribution));
```

Comment:
SInce those function are internal, it is fine to have them return the smaller (and correct) types. The places where we want to keep using uint256 are in the external interfaces, where we simply upcast.

In other words, keep the smaller type internally everywhere, only upcast when exposing it in external getters.

Reply:
Are you sure? I'd say that the opposite is preferable. Use the "true" types for the external interface to take advantage of the automatic max value checking but use `uint256` internally where we know that the values are safe to keep gas costs low.